pravega-client-examples/src/main/java/io/pravega/example/secure/SecureBatchReader.java

```diff
@@ -0,0 +1,319 @@
+    private ClientConfig clientConfig;
+
+    private static Logger LOG = LoggerFactory.getLogger(SecureBatchReader.class);
+    private static final int THREAD_POOL_SIZE = 5;
+    private static ScheduledExecutorService streamCutLoggerService = newScheduledThreadPool(THREAD_POOL_SIZE, String.format("EventsCount"));
+    private ScheduledExecutorService executor;
+
+
+    public SecureBatchReader(String scope, String stream, URI controllerURI,
+                        String truststorePath, boolean validateHostname,
+                        String username, String password) {
+
+        this.scope = scope;
+        this.stream = stream;
+        this.controllerURI = controllerURI;
+        this.truststorePath = truststorePath;
+        this.validateHostName = validateHostname;
+        this.username = username;
+        this.password = password;
+    }
+
+
+    public void read() throws ReinitializationRequiredException {
+
+        /*
+         * Note about setting the client config for HTTPS:
+         *    - The client config below is configured to use an optional truststore. The truststore is expected to be
+         *      the certificate of the certification authority (CA) that was used to sign the server certificates.
+         *      If this is null or empty, the default JVM trust store is used. In this demo, we use a provided
+         *      "cert.pem" as the CA certificate, which is also provided on the server-side. If the cluster uses a
+         *      different CA (which it should), use that CA's certificate as the truststore instead.
+         *
+         *    - Also, the client config below disables host name verification. If the cluster's server certificates
+         *      have DNS names / IP addresses of the servers specified in them, you may turn this on. In a production
+         *      deployment, it is recommended to keep this on.
+         *
+         * Note about setting the client config for auth:
+         *    - The client config below is configured with an object of DefaultCredentials class. The user name
+         *      and password arguments passed to the object represent the credentials used for authentication
+         *      and authorization. The assumption we are making here is that the username is valid on the server,
+         *      the password is correct and the username has all the permissions necessary for performing the
+         *      subsequent operations.
+         */
+        clientConfig = ClientConfig.builder()
+                .controllerURI(this.controllerURI) // "tls://localhost:9090"
+
+                // TLS-related client-side configuration
+                .trustStore(this.truststorePath)
+                .validateHostName(this.validateHostName)
+
+                // Auth-related client-side configuration
+                .credentials(new DefaultCredentials(this.password, this.username))
+                .build();
+
+        System.out.println("Done creating a client config.");
```

Comment:
I think that we should be using `log` instead of `System.out` or `System.err` whenever possible. Can you please check if we use this kind of output normally in the rest of Pravega samples?

Reply:
Yes Andre